### PR TITLE
Add RRFS-A to weekly tests

### LINKED ISSUE #1417: Add RRFS-A to weekly tests
## Description
Since RRFS-A is a candidate for implementation, we need to test high resolution, large domain configuration. We plan on testing it on Hera first.

## Solution
Add new test in rt_weekly.conf

## Alternatives
N/A

### PR DESCRIPTION
**Pull request by RatkoVasic-NOAA**
# PR Checklist

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR
are specified below.
Issue #1417 

- [-] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [+] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.
 Already added it to Hera:
 /scratch1/NCEPDEV/nems/emc.nemspara/RT/NEMSfv3gfs/input-data-20220414/fv3_regional_rrfs_a/INPUT/
(need to propagate to all other machines, although not in use)
**Code managers please synchronize input-data-20220414 from Hera to all other supported machines.**

## Instructions: All subsequent sections of text should be filled in as appropriate.

Adding new test, can be merged with other PR's. Only one file changed (rt_weekly.conf) and 6 new files added to tests directory.

## Description

Adding new test to Hera machine, weekly tests

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!)
- Closes #1417 

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [ ] hera.gnu
- [ ] orion.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [ ] gaea.intel
- [ ] jet.intel
- [ ] wcoss2.intel
- [ ] acorn.intel
- [ ] opnReqTest for newly added/changed feature
- [ ] CI

### CODE REVIEW

File: tests/fv3_conf/regional_rrfs_a_run.IN [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+cp @[INPUTDATA_ROOT]/FV3_input_data_INCCN_aeroclim/MERRA2/merra2.aerclim.2003-2014.m01.nc aeroclim.m03.nc
+cp @[INPUTDATA_ROOT]/FV3_input_data_INCCN_aeroclim/MERRA2/merra2.aerclim.2003-2014.m01.nc aeroclim.m04.nc
+cp @[INPUTDATA_ROOT]/FV3_input_data_INCCN_aeroclim/MERRA2/merra2.aerclim.2003-2014.m01.nc aeroclim.m05.nc
+cp @[INPUTDATA_ROOT]/FV3_input_data_INCCN_aeroclim/MERRA2/merra2.aerclim.2003-2014.m01.nc aeroclim.m06.nc
+cp @[INPUTDATA_ROOT]/FV3_input_data_INCCN_aeroclim/MERRA2/merra2.aerclim.2003-2014.m01.nc aeroclim.m07.nc
+cp @[INPUTDATA_ROOT]/FV3_input_data_INCCN_aeroclim/MERRA2/merra2.aerclim.2003-2014.m01.nc aeroclim.m08.nc
+cp @[INPUTDATA_ROOT]/FV3_input_data_INCCN_aeroclim/MERRA2/merra2.aerclim.2003-2014.m01.nc aeroclim.m09.nc
+cp @[INPUTDATA_ROOT]/FV3_input_data_INCCN_aeroclim/MERRA2/merra2.aerclim.2003-2014.m01.nc aeroclim.m10.nc
+cp @[INPUTDATA_ROOT]/FV3_input_data_INCCN_aeroclim/MERRA2/merra2.aerclim.2003-2014.m01.nc aeroclim.m11.nc
+cp @[INPUTDATA_ROOT]/FV3_input_data_INCCN_aeroclim/MERRA2/merra2.aerclim.2003-2014.m01.nc aeroclim.m12.nc
```
Comment (DusanJovic-NOAA) [inline tests/fv3_conf/regional_rrfs_a_run.IN]: Is this correct? Same file copied 12 times into 12 different files.

File: tests/parm/regional_rrfs_a.nml.IN [hunk 2]
```diff
@@ -0,0 +1,314 @@
+&amip_interp_nml
+    data_set = 'reynolds_oi'
+    date_out_of_range = 'climo'
+    interp_oi_sst = .true.
+    no_anom_sst = .false.
+    use_ncep_ice = .false.
+    use_ncep_sst = .true.
+/
+
```
Comment (DusanJovic-NOAA) [inline tests/parm/regional_rrfs_a.nml.IN]: `amip_interp_nml` namelist is not used by the model. Should be removed.

File: tests/parm/model_configure_regional_rrfs_a.IN [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+cpl:                     .false.
+calendar:                'julian'
+memuse_verbose:          .false.
+atmos_nthreads:          2
+use_hyper_thread:        .false.
+ncores_per_node:         40
+debug_affinity:          .true.
+restart_interval:        @[RESTART_INTERVAL]
+output_1st_tstep_rst:    .false.
+print_esmf:              .false.
```
Comment (DusanJovic-NOAA) [inline tests/parm/model_configure_regional_rrfs_a.IN]: `total_member`, `PE_MEMBER01`, `RUN_CONTINUE`, `ENS_SPS`, `memuse_verbose`, `atmos_nthreads`, `use_hyper_thread`, `ncores_per_node`, `debug_affinity`, `print_esmf` are not used (anymore) by the model.

File: tests/parm/model_configure_regional_rrfs_a.IN [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+#
+# Write-component (quilting) computational parameters.
+#
+write_groups:            @[WRITE_GROUP]
+write_tasks_per_group:   @[WRTTASK_PER_GROUP]
+num_files:               2
+filename_base:           'dyn''phy'
+output_file:             'netcdf'
+write_nemsioflip:        .false.
+write_fsyncflag:         .false.
```
Comment (DusanJovic-NOAA) [inline tests/parm/model_configure_regional_rrfs_a.IN]: `write_fsyncflag` is not used anymore.

File: tests/fv3_conf/regional_rrfs_a_run.IN (hunk 1 above)
Comment (RatkoVasic-NOAA) [reply tests/fv3_conf/regional_rrfs_a_run.IN]: Corrected

File: tests/parm/regional_rrfs_a.nml.IN (hunk 2 above)
Comment (RatkoVasic-NOAA) [reply tests/parm/regional_rrfs_a.nml.IN]: corrected

File: tests/parm/model_configure_regional_rrfs_a.IN (hunk 3 above)
Comment (RatkoVasic-NOAA) [reply tests/parm/model_configure_regional_rrfs_a.IN]: removed

File: tests/parm/model_configure_regional_rrfs_a.IN (hunk 4 above)
Comment (RatkoVasic-NOAA) [reply tests/parm/model_configure_regional_rrfs_a.IN]: removed

**Comment by jkbk2004** [comment]
@RatkoVasic-NOAA If possible, can you create pr to @DusanJovic-NOAA 's fork branch to combine this pr to #1434 ?

**Comment by jkbk2004** [comment]
@RatkoVasic-NOAA ignore my previous message. We will commit this PR on Monday.

**Comment by jkbk2004** [comment]
@BrianCurtis-NOAA a new INPUT directory was added on Hera under /scratch1/NCEPDEV/nems/emc.nemspara/RT/NEMSfv3gfs/input-data-20220414/fv3_regional_rrfs_a @RatkoVasic-NOAA  This new input file will be used in weekly test. So no baseline change to existing regression tests, right?

**Comment by BrianCurtis-NOAA** [comment]
@RatkoVasic-NOAA Do you have a log file of this RT passing?

@jkbk2004 I am not sure this needs full RT testing if we're just adding a test to the weekly RT.

**Comment by RatkoVasic-NOAA** [comment]
> @RatkoVasic-NOAA Do you have a log file of this RT passing?
> 
> @jkbk2004 I am not sure this needs full RT testing if we're just adding a test to the weekly RT.

No, weekly test results (logs) are not stored in repository.

**Comment by jkbk2004** [comment]
I think we can do a quick sanity run with a machine like orion. Once we confirm with the sanity run, we should be ok to merge in.

**Comment by BrianCurtis-NOAA** [comment]
@RatkoVasic-NOAA i think `./rt.sh -e -n regional_rrfs_a > rtn.out 2>&1 &` might work? I assume you've run it successfully beforehand, so Jong's sanity check should be OK.

**Comment by RatkoVasic-NOAA** [comment]
Yes, it runs on Hera only.

**Comment by BrianCurtis-NOAA** [comment]
> I think we can do a quick sanity run with a machine like orion. Once we confirm with the sanity run, we should be ok to merge in.

Run on Hera

**Comment by RatkoVasic-NOAA** [comment]
I just let it run on Hera and it is failing in compilation:
```
CMake Error at CMakeLists.txt:119 (find_package):
  By not providing "FindNetCDF.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "NetCDF", but
  CMake did not find one.

  Could not find a package configuration file provided by "NetCDF" (requested
  version 4.7.4) with any of the following names:

    NetCDFConfig.cmake
    netcdf-config.cmake

  Add the installation prefix of "NetCDF" to CMAKE_PREFIX_PATH or set
  "NetCDF_DIR" to a directory containing one of the above files.  If "NetCDF"
  provides a separate development package or SDK, be sure it has been
  installed.
```

**Comment by RatkoVasic-NOAA** [comment]
@BrianCurtis-NOAA @jkbk2004 
I ran it from old directory (non updated). Now test IS running on Hera,
Hera:/scratch2/NCEPDEV/fv3-cam/Ratko.Vasic/wrk/RRFS/yyy/tests

**Comment by jkbk2004** [comment]
rsynced ok new input files of fv3_regional_rrfs_a to all machines

**Comment by RatkoVasic-NOAA** [comment]
@jkbk2004 @BrianCurtis-NOAA 
Test passed on Hera:
```
Hera:/scratch2/NCEPDEV/fv3-cam/Ratko.Vasic/wrk/RRFS/yyy/tests>cat RegressionTests_weekly_hera.intel.log
Mon Oct  3 20:07:21 UTC 2022
Start Regression test

Compile 001 elapsed time 321 seconds. -DAPP=ATM -DCCPP_SUITES=FV3_HRRR -D32BIT=ON -DMPI=ON -DCMAKE_BUILD_TYPE=Release

  0: The total amount of wall time                        = 540.111654
  0: The maximum resident set size (KB)                   = 3168296

Test 001 regional_rrfs_a RUN_SUCCESS


REGRESSION TEST WAS SUCCESSFUL
Mon Oct  3 20:33:57 UTC 2022
Elapsed time: 00h:26m:36s. Have a nice day!
```

**Comment by jkbk2004** [comment]
All sounds good. This PR is ready.

**Review by jkbk2004 — APPROVED** [review]

**Comment by jkbk2004** [comment]
@binli2337 @BrianCurtis-NOAA We can start merging in this pr. I will close issue #1417. Go ahead to leave final review and approve.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Comment by BrianCurtis-NOAA** [comment]
> @binli2337 @BrianCurtis-NOAA We can start merging in this pr. I will close issue #1417. Go ahead to leave final review and approve.

@jkbk2004 you can use the top of the PR to add a "closes #1417" so the PR will automatically close that issue once merged.

**Pull request merged**